packages/slate-react/src/plugin/react-editor.ts

```diff
@@ -778,6 +793,35 @@ export const ReactEditor: ReactEditorInterface = {
             offset -= el.textContent!.length
           })
         }
+      } else if (nonEditableNode) {
+        // Find the edge of the nearest leaf in `searchDirection` from the
+        // non-editable node
+        const leafNodes = Array.from(
+          editorEl.querySelectorAll(
+            // This editor's leaves (not those of a nested editor)
+            '[data-slate-leaf]:not(:scope [data-slate-editor] [data-slate-leaf])'
+          )
+        )
+        leafNode =
+          (searchDirection === 'forward'
+            ? leafNodes.find(leaf => isAfter(nonEditableNode, leaf))
+            : leafNodes.findLast(leaf => isBefore(nonEditableNode, leaf))) ??
+          null
+
+        if (!leafNode) {
+          offset = 1
+        } else {
```

Comment:
I copied this from the `else if (voidNode)` block, which prefixes it with `// COMPAT: In read-only editors the leaf is not rendered.` But I'm thinking that may not be applicable here. I'm considering replacing this with

```suggestion
        if (leafNode) {
```

Reply:
Done in https://github.com/ianstormtaylor/slate/pull/5716/commits/1f368a9748298dac1ed8870031d2268ae14011e0